tracing-subscriber/src/filter/env/directive.rs

```diff
@@ -237,6 +211,47 @@ impl FromStr for Directive {
     }
 }
 
+impl Match for Directive {
+    fn cares_about(&self, meta: &Metadata<'_>) -> bool {
+        // Does this directive have a target filter, and does it match the
+        // metadata's target?
+        if let Some(ref target) = self.target {
+            if !meta.target().starts_with(&target[..]) {
+                return false;
+            }
+        }
+
+        // Do we have a name filter, and does it match the metadata's name?
+        // TODO(eliza): put name globbing here?
+        if let Some(ref name) = self.in_span {
+            if name != meta.name() {
+                return false;
+            }
+        }
+
+        // Does the metadata define all the fields that this directive cares about?
+        let fields = meta.fields();
```

Comment:
nit: i'd maybe call `fields` something else to more clearly communicate that the fields are coming from the metadata for the span/event. i might just be getting dumber, but it took me a second to realize that `fields.field` is not the same as `self.field`.

Reply:
i'm fine with changing this, but this isn't new code --- the names were already there. GitHub just thinks this is "new" because other code was inserted above it: https://github.com/tokio-rs/tracing/pull/2035/files/7b099fbe261973e33f47b708738052b965743043#diff-94873ee31f65c84ce0c17d7d69c8dfe667bc2b0821fcc404f04ffe7c4d009bffL112-L134

Reply:
whoops! i knew it looked familiar.